core/src/test/java/org/apache/iceberg/hadoop/TestCachingCatalog.java

```diff
@@ -257,6 +264,44 @@ public void testCacheExpirationEagerlyRemovesMetadataTables() throws IOException
             catalog.cache().asMap().containsKey(metadataTable)));
   }
 
+  @Test
+  @Ignore("reproduces https://github.com/apache/iceberg/issues/3791")
```

Comment:
This test should pass now right, so should we turn this on?

Reply:
That's an open question. I think we should enable it, as it runs relatively fast (~2 seconds) and we would be getting the safety of not hitting the deadlock issue again.

I guess @kbendick thinks differently though

Reply:
I figured to turn it off normally since it uses a lot of threads. I was hoping that we might come up with a system of tagging tests as resource intensive etc that we run nightly or something.

I've seen a handful of tests lately that start many threads and that seems not great for CI.

But if we're ok with running this on every push, we can remove the `Ignore`. We could either open an issue for expensive tests that we flag or hold off on that as we might not be there yet.

But tests that spawn 20+ threads make me nervous about increasing oddities in CI like HMS timeouts etc.

Reply:
If it's just 2 seconds, I'd say enable it. We share infra with all of ASF,  but arguably we have a lot of Spark tests that should be refactored that are more resource intensive than this. The risk of thread thrashing is likely minimal and we can deal with it if it comes up.